doc/c-style-guide.md

```diff
@@ -0,0 +1,41 @@
+GRPC C STYLE GUIDE
+=====================
+
+Background
+----------
+
+Here we document style rules for C usage in the gRPC Core library.
+
+General
+-------
+
+- Layout rules are defined by clang-format, and all code should be passed through
+  clang-format. A (docker-based) script to do so is included in 
+  tools/distrib/clang_format_code.sh.
+
+Header Files
+------------
+
+- Public header files (those in the include/grpc tree) should compile as pedantic C89
```

Comment:
Additional tidbit in slight favor of C89 pedantry:
`vspython27` (the semi-pseudo-official Python extension compiler for Windows) is stuck at Visual Studio version 9, or the 2008 version; this was back when C99 was entirely unsupported in Visual Studio (sans going cheatyface and compiling as C++).